python_modules/dagster/dagster/_config/pythonic_config/__init__.py

```diff
@@ -200,15 +200,26 @@ def __init__(self, **config_dict) -> None:
         for key, value in config_dict.items():
             field = self.__fields__.get(key)
             if field and field.field_info.discriminator:
-                nested_items = list(check.is_dict(value).items())
+                nested_dict = value
+
+                discriminator_key = check.not_none(field.discriminator_key)
+                if isinstance(value, Config):
+                    value_dict = value._as_config_dict()  # noqa: SLF001
+                    value_discriminator = check.not_none(value_dict.get(discriminator_key))
+                    values_without_key = {
+                        k: v for k, v in value_dict.items() if k != discriminator_key
+                    }
+                    nested_dict = {value_discriminator: values_without_key}
+
+                nested_items = list(check.is_dict(nested_dict).items())
```

Comment:
shouldn't this logic reside inside of `_as_config_dict`?

Reply:
The current name doesn't adequately describe what the method does (a shallow flattening of a config class to a dict, a user might expect a deep conversion).  I'm moving the logic into a method in the config class and doing some slight restructuring/renaming to match.